src/Features/Core/Portable/ExtractMethod/MethodExtractor.VariableSymbol.cs

```diff
@@ -21,294 +17,172 @@ internal abstract partial class AbstractExtractMethodService<
+                    _ when (this._displayOrder != other._displayOrder) => this._displayOrder - other._displayOrder,
+                    // Finally, compare within the general class of the variable.
+                    _ => this.CompareToWorker(other),
+                };
             }
-        }
-
-        protected abstract class NotMovableVariableSymbol(ITypeSymbol type) : VariableSymbol(type)
-        {
-            public override SyntaxToken GetOriginalIdentifierToken(CancellationToken cancellationToken)
-                => default;
 
-            public override SyntaxAnnotation IdentifierTokenAnnotation => throw ExceptionUtilities.Unreachable();
-
-            public override void AddIdentifierTokenAnnotationPair(
-                MultiDictionary<SyntaxToken, SyntaxAnnotation> annotations, CancellationToken cancellationToken)
-            {
-                // do nothing for parameter
-            }
+            public string Name => this.GetSymbol().ToDisplayString(
+                new SymbolDisplayFormat(
+                    parameterOptions: SymbolDisplayParameterOptions.IncludeName,
+                    miscellaneousOptions: SymbolDisplayMiscellaneousOptions.EscapeKeywordIdentifiers));
         }
 
-        protected sealed class ParameterVariableSymbol : NotMovableVariableSymbol, IComparable<ParameterVariableSymbol>
+        protected abstract class VariableSymbol<TVariableSymbol, TSymbol>(
+            TSymbol symbol, ITypeSymbol symbolType, int displayOrder)
+            : VariableSymbol(symbolType, displayOrder)
+            where TVariableSymbol : VariableSymbol<TVariableSymbol, TSymbol>
+            where TSymbol : ISymbol
         {
-            private readonly IParameterSymbol _parameterSymbol;
+            protected TSymbol Symbol { get; } = symbol;
 
-            public ParameterVariableSymbol(IParameterSymbol parameterSymbol, ITypeSymbol type)
-                : base(type)
-            {
-                Contract.ThrowIfNull(parameterSymbol);
-                _parameterSymbol = parameterSymbol;
-            }
+            protected override ISymbol GetSymbol() => this.Symbol;
 
-            public override int DisplayOrder => 0;
+            protected sealed override int CompareToWorker(VariableSymbol right)
+                => this.CompareTo((TVariableSymbol)right);
 
-            protected override int CompareTo(VariableSymbol right)
-                => CompareTo((ParameterVariableSymbol)right);
+            protected abstract int CompareTo(TVariableSymbol right);
 
-            public int CompareTo(ParameterVariableSymbol other)
+            protected static int DefaultCompareTo(ISymbol left, ISymbol right)
             {
-                Contract.ThrowIfNull(other);
-
-                if (this == other)
-                {
-                    return 0;
-                }
+                var locationLeft = left.Locations.First();
```

Comment:
>.First() [](http://example.com/codeflow?start=49&length=8)

nit: I like [0] better, makes it obvious that it's not needing to start an enumeration as the type of Locations isn't visible.

Reply:
fair.